File: 단계별로 풀어보기/Step 07/1065.cpp
#include <iostream>
using namespace std;

bool han(int n) {
    if(n<100) return true;
    else if(n>=100) {
        int x = n/100-(n/10)%10;
        int y = (n/10)%10-n%10;
        if(x==y) return true;
        else return false;
    }
}

int main() {
    int n;
    cin >> n;

    int cnt = 0;
    int i;
    for(i=1; i<=n; i++) {
        if(han(i)) cnt++;
    }
    cout << cnt;
    return 0;
}

File: StepByStep/Python/Step_05_함수/1065.py
def han(n):
    if n < 100:
        return n
    else:
        cnt = 99
        for i in range(100, n+1):
            num = str(i)
            if int(num[0]) - int(num[1]) == int(num[1]) - int(num[2]):
                cnt += 1
        return cnt



N = int(input())
print(han(N))